core/src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -72,21 +72,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                     // skip
                 } else if (token == XContentParser.Token.START_ARRAY) {
-                    if ("filters".equals(currentFieldName)) {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
-                        }
-                    } else {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
+                    queriesFound = true;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        Query filter = parseContext.parseInnerFilter();
+                        if (filter != null) {
+                            queries.add(filter);
```

Comment:
I wonder if we should be a bit more lenient around what is allowed and what isn't... the idea is that you can wrap the array in a `filters` element, if the `and` filter needs to contain more than just the filters, or simply skip that `filters` element when not needed. Right now (and before this change too) we accept any element I think e.g. whatever_filters. Shall we change that?